### ISSUE TITLE
gradlew 6.6 does not work when path contains double quote

### ISSUE THREAD

**Issue by vlsi** [gradle/gradle#14708]
### Expected Behavior

The project should be buildable even in case the folder contains double quote

### Current Behavior

```
$ pwd
.../calcite " +
$ ./gradlew classes
Error: Could not find or load main class +.media.gradle.icns -Dorg.gradle.appname=gradlew -classpath .Users.vladimirsitnikov.Documents.code.calcite
Caused by: java.lang.ClassNotFoundException: +.media.gradle.icns -Dorg.gradle.appname=gradlew -classpath .Users.vladimirsitnikov.Documents.code.calcite
```

### Context

I'm updating tasks to properly escape file paths

### Steps to Reproduce 

Use space with double quote for Gradle proeject

### Your Environment

macOS 10.14.6
Build scan URL: 🤷‍♂️

**Comment by bmuskalla** [comment]
Closing as dup of #15977

**Comment by vlsi** [comment]
Should #15977 be closed instead?

**Comment by bmuskalla** [comment]
I closed this one because #15977 already captured multiple scenarios where this fails (not only double quotes)

**Comment by mlopatkin** [comment]
#15977 turned out to be Windows-specific, so I'm reopening this one instead.

**Comment by mlopatkin** [comment]
Fixed in #16055.
